test/unit/mdc-data-table/mdc-data-table.test.js

```diff
@@ -279,76 +255,13 @@ test('adapter#setAttributeAtRowIndex', () => {
   component.destroy();
 });
 
-test('adapter#registerRowCheckboxes Initiates all row checkboxes', () => {
-  const {
-    mockCheckboxFactory,
-    mockFoundation,
-    root,
-    component,
-    adapter,
-  } = setupTestWithMocks();
-
-  const rows = [].slice.call(root.querySelectorAll(strings.ROW_SELECTOR));
-  td.when(mockFoundation.getRows()).thenReturn(rows);
-
-  adapter.registerRowCheckboxes();
-
-  td.verify(mockCheckboxFactory(td.matchers.isA(Element)), {times: 3});
```

Comment:
isn't this test still valid with the exception of the component.destroy() call?

Reply:
It is a valid test but mocking isn't helping much. But, I discovered more bugs without mocking compared to with mocking.